test/suite/wtbackup.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python
+#
+# Public Domain 2014-2021 MongoDB, Inc.
+# Public Domain 2008-2014 WiredTiger, Inc.
+#
+# This is free and unencumbered software released into the public domain.
+#
+# Anyone is free to copy, modify, publish, use, compile, sell, or
+# distribute this software, either in source code form or as a compiled
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+import os
+import wttest
+from suite_subprocess import suite_subprocess
+from helper import compare_files
+
+# Shared base class used by backup tests.
+class backup_base(wttest.WiredTigerTestCase, suite_subprocess):
+    #
+    # Add data to the given uri.
+    # options:
+    #   cursor_options: a config string for cursors
+    #   num_ops: number of operations added to uri
+    #   mult: counter to have variance in data
+    #   counter: a counter to used to produce unique backup ids
+    #   session_checkpoint: boolean in which enables session checkpoints after adding data
+    #   initial_backup: To determine whether to increase/decrease counter, which determines
```

Comment:
I feel like we can remove **initial_backup** from here, and the backup tests that use this variable. Instead an option would be to increase the counter if the test needs it to, by assigning to the return value of the **add_data** function. @sueloverso  what are your thoughts?

Reply:
@ddanderson can you take a look at this PR and the use of `data_options`? Do you have any suggestions on the best way to refactor and use these arguments? Does it overlap with the `scenarios` or other underlying Python functions we already have?